### ISSUE TITLE
Tidal process should survive parser errors

### ISSUE THREAD

**Issue by yaxu** [tidalcycles/Tidal#432]
See e.g. #431 
This was supposed to have been solved by https://github.com/tidalcycles/Tidal/commit/e0ccfd3fe2a491730ca677a048b2a3ac907b683c

**Comment by yaxu** [comment]
For example:
```
d1 $ s "drum*8" # n "0..8"
```

**Comment by yaxu** [comment]
Strangely, this errors fine:
```
d1 $ sound "bd" +| n "0..8"
```
but this errors too late:
```
d1 $ sound "bd" +| n "0..8"
```

**Comment by tomasraposo** [comment]
is there any way then of not having to reopen atom every time there is an error? thanks. 

or is it a matter of its new  syntax, which forcefully presupposes good writing at all times?

**Comment by yaxu** [comment]
@tomasraposo It's a bug, I'll have it fixed soon!

**Comment by yaxu** [comment]
In this the "0..8" never gets evaluated, even though the `seq` is there:
```
let x = 0 |> "0..8" :: Pattern Double
in
putStrLn $ seq x "hello"
```
However replacing `|>` with `>|`, and a parse exception ''is'' raised, as expected
```
let x = 0 |> "0..8" :: Pattern Double
in
putStrLn $ seq x "hello"
```
Puzzling!

**Comment by tomasraposo** [comment]
i see now.  puzzling indeed. now it makes sense. thanks @yaxu! appreciated.

**Comment by tomasraposo** [comment]
i've just updated to 1.0.1 and i still keep getting the same error...

I really don't understand why is this happening... :/

**Comment by yaxu** [comment]
@tomasraposo can you share a pattern that causes the error?

**Comment by yaxu** [comment]
Ok I've released 1.0.2 now, hopefully will fix it..

**Comment by tomasraposo** [comment]
@echophon @yaxu

it fixes the issue indeed. now atom doesn't crash whenever i make a typo. great! 

thanks alex. now back to coding :)!

**Comment by diegodorado** [comment]
That was quick! thanks! This also fixes #433

**Comment by yaxu** [comment]
@mhetrick I can't reproduce that problem, and am going to close this now, but please feel free to reopen this issue if it happens again.
